buildkite/client/pipeline.go

```diff
@@ -85,56 +93,32 @@ func (c *Client) CreatePipeline(pipeline *Pipeline) (*Pipeline, error) {
 	return &result, nil
 }
 
-// createPipelineGraphQl will create the pipeline but only set the required fields
-// after creation, UpdatePipeline will be used to set the rest of the fields via
-// the REST API
-func (c *Client) createPipelineGraphQl(pipeline *Pipeline) (*Pipeline, error) {
-	req := graphql.NewRequest(`
-mutation PipelineCreateRequest($pipelineCreateInput: PipelineCreateInput!) {
-  pipelineCreate(input: $pipelineCreateInput) {
-    pipeline {
-      slug
-    }
-  }
-}`)
-
-	orgID, err := c.GetOrganizationId(c.orgSlug)
-	if err != nil {
-		return nil, err
-	}
-
-	req.Var("pipelineCreateInput", map[string]interface{}{
-		"organizationId": orgID,
-		"name":           pipeline.Name,
-		"repository": map[string]string{
-			"url": pipeline.Repository,
-		},
-		"steps": map[string]string{
-			"yaml": pipeline.Configuration,
+// createYAMLPipeline will create the pipeline but only set the required fields
+// filled with stubs. After creation, UpdatePipeline will be used to set the rest
+// of the fields via the REST API
+func (c *Client) createYAMLPipeline(pipeline *Pipeline) (*Pipeline, error) {
+	created, err := c.CreatePipeline(&Pipeline{
+		Name:       pipeline.Name,
+		Repository: pipeline.Repository,
+		TeamUUIDs:  pipeline.TeamUUIDs,
+		Steps: []Step{
+			{
+				Type:    "script",
+				Name:    "Script",
+				Command: "command.sh",
+			},
```

Comment:
Both versions, REST and GraphQL support team uuids, on pipeline creation. But *only* GraphQL supports providing a yaml spec as steps definition, so please don't change this. It's intended. 

Check the [GraphQL Explorer](https://graphql.buildkite.com/explorer) for documentation:
![Screenshot_2020-01-17 Buildkite GraphQL Explorer Beta](https://user-images.githubusercontent.com/1070051/72593139-83ac9700-3904-11ea-824c-a0b9cf762c52.png)


And why overwrite the speps with a static value?

Reply:
> Both versions, REST and GraphQL support team uuids.

REST API supports `team_uuids` array field and GraphQL supports an array of POJOs with `id` and `accessLevel` in it so APIs are different and I've changed pipeline creation from GraphQL to REST because of this. The rest logic<sup>[[1]](https://github.com/tink-ab/terraform-buildkite/pull/11/files#diff-21be118a3b5361319e14e7dea1a3049eL132)</sup> is the same as it was. So we still use create step and update step with properties that create methods not support. As for static value, REST API requires at least one step defined so I do it in this way. The step is being overridden right after the pipeline was updated. In both cases, the pipeline state between create and update method calls is potentially inconsistent anyway so don't see a significant difference between creating pipeline methods.

As per my comments above, I am not going to create teams' management in the pipeline definition body. Moreover, I would keep it as simple as possible in terms of usability for the user.

I understand your concerns about switching pipeline creation back to REST API and would keep it in as modern and fancy way as possible but in this case, to reduce the complexity and get things done I think it's a good trade-off especially because it doesn't bring any degradations in terms of available features to use.

If this PR applies any kind of limitations please let me know, I am happy to address them.

[1] - https://github.com/tink-ab/terraform-buildkite/pull/11/files#diff-21be118a3b5361319e14e7dea1a3049eL132